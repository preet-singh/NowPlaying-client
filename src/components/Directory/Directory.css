@media screen and (min-width: 350px){
    .Directory {
        color: white;
        padding: 20px;
        text-align: center;
    }

    .directory-dropdown {
        padding: 10px;
    }
}