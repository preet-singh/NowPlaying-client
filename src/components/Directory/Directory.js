//Dependencies
import React from 'react';
import {withRouter, Link} from 'react-router-dom';

//Components
import SearchBar from '../SearchBar/SearchBar';
import UserContext from '../../utils/context';
import AuthService from '../../utils/auth-service';


class Directory extends React.Component {
  static contextType = UserContext;
  constructor(props) {
    super(props);
    this.state = {
    }
  }
  
  getCategoryList = (categoryList, category) => {
    let categoryListNames = categoryList.map(item => item.media_type) || [];
    let selectOptions = [];
    categoryListNames.forEach((item,index) => {
        selectOptions.push(<option value={item} key={item}>{item}</option>)
    });
    return selectOptions;
  }

  render() {
    let categoryList = this.context.categoryList || [];
    let category = this.context.category || ''
    return(
      <div className="Directory">
        <section className="directory-box">
          <div className="directory-dropdown">
            <select onChange={(e) => this.context.setCategory(e.target.value)}>
              {this.getCategoryList(categoryList, category)}
            </select>
          </div>
        </section>
        <SearchBar />
      </div>
    );
  }
}

export default withRouter(Directory);