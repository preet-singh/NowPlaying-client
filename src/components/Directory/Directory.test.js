import React from 'react';
import ReactDOM from 'react-dom';
import Directory from './Directory';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Directory testing', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <MemoryRouter>
            <Directory />
        </MemoryRouter>, 
    div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders the categories within the directory', () => {
    const wrapper = shallow(<Directory />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  
});