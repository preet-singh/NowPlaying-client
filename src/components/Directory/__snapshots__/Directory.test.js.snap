// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory testing renders the categories within the directory 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
